File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/cache/model/UserCache.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.cache.model;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import com.zfoo.net.packet.common.PairLS;
import com.zfoo.protocol.IPacket;

import java.util.List;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2019-11-08 11:00
 */
public class UserCache implements IPacket {

    public static final transient short PROTOCOL_ID = 3000;

    @JsonSerialize(using = ToStringSerializer.class)
    private long id;
    private String name;
    @JsonSerialize(using = ToStringSerializer.class)
    private long coin;
    private String avatar;
    private String background;
    private int gender;
    private String signature;
    @JsonSerialize(using = ToStringSerializer.class)
    private long customTime;
    private String custom;
    private int fanNum;
    private int followNum;
    private int starNum;
    @JsonSerialize(using = ToStringSerializer.class)
    private long free;
    @JsonSerialize(using = ToStringSerializer.class)
    private long normal;
    private List<PairLS> locations;
    private List<PairLS> persons;
    private List<PairLS> items;

    public static UserCache valueOf(long id, String name) {
        var cache = new UserCache();
        cache.id = id;
        cache.name = name;
        return cache;
    }

    public static UserCache valueOf(long id, String name, long coin, String avatar, String background, int gender, String signature
            , long customTime, String custom
            , int followNum, int fanNum, int starNum, long free, long normal
            , List<PairLS> locations, List<PairLS> persons, List<PairLS> items) {
        var cache = new UserCache();
        cache.id = id;
        cache.name = name;
        cache.coin = coin;
        cache.avatar = avatar;
        cache.background = background;
        cache.gender = gender;
        cache.signature = signature;
        cache.customTime = customTime;
        cache.custom = custom;
        cache.followNum = followNum;
        cache.fanNum = fanNum;
        cache.starNum = starNum;
        cache.free = free;
        cache.normal = normal;
        cache.locations = locations;
        cache.persons = persons;
        cache.items = items;
        return cache;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getCoin() {
        return coin;
    }

    public void setCoin(long coin) {
        this.coin = coin;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getBackground() {
        return background;
    }

    public void setBackground(String background) {
        this.background = background;
    }

    public int getGender() {
        return gender;
    }

    public void setGender(int gender) {
        this.gender = gender;
    }

    public String getSignature() {
        return signature;
    }

    public void setSignature(String signature) {
        this.signature = signature;
    }

    public long getCustomTime() {
        return customTime;
    }

    public void setCustomTime(long customTime) {
        this.customTime = customTime;
    }

    public String getCustom() {
        return custom;
    }

    public void setCustom(String custom) {
        this.custom = custom;
    }

    public int getFollowNum() {
        return followNum;
    }

    public void setFollowNum(int followNum) {
        this.followNum = followNum;
    }

    public int getFanNum() {
        return fanNum;
    }

    public void setFanNum(int fanNum) {
        this.fanNum = fanNum;
    }

    public int getStarNum() {
        return starNum;
    }

    public void setStarNum(int starNum) {
        this.starNum = starNum;
    }

    public long getFree() {
        return free;
    }

    public void setFree(long free) {
        this.free = free;
    }

    public long getNormal() {
        return normal;
    }

    public void setNormal(long normal) {
        this.normal = normal;
    }

    public List<PairLS> getLocations() {
        return locations;
    }

    public void setLocations(List<PairLS> locations) {
        this.locations = locations;
    }

    public List<PairLS> getPersons() {
        return persons;
    }

    public void setPersons(List<PairLS> persons) {
        this.persons = persons;
    }

    public List<PairLS> getItems() {
        return items;
    }

    public void setItems(List<PairLS> items) {
        this.items = items;
    }
}

File: web/src/jsProtocol/cache/model/UserCache.js
// @author jaysunxiao
// @version 1.0
// @since 2019-11-08 11:00
const UserCache = function(avatar, background, coin, custom, customTime, fanNum, followNum, free, gender, id, items, locations, name, normal, persons, signature, starNum) {
    this.avatar = avatar; // java.lang.String
    this.background = background; // java.lang.String
    this.coin = coin; // long
    this.custom = custom; // java.lang.String
    this.customTime = customTime; // long
    this.fanNum = fanNum; // int
    this.followNum = followNum; // int
    this.free = free; // long
    this.gender = gender; // int
    this.id = id; // long
    this.items = items; // java.util.List<com.zfoo.net.packet.common.PairLS>
    this.locations = locations; // java.util.List<com.zfoo.net.packet.common.PairLS>
    this.name = name; // java.lang.String
    this.normal = normal; // long
    this.persons = persons; // java.util.List<com.zfoo.net.packet.common.PairLS>
    this.signature = signature; // java.lang.String
    this.starNum = starNum; // int
};

UserCache.prototype.protocolId = function() {
    return 3000;
};

UserCache.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writeString(packet.avatar);
    buffer.writeString(packet.background);
    buffer.writeLong(packet.coin);
    buffer.writeString(packet.custom);
    buffer.writeLong(packet.customTime);
    buffer.writeInt(packet.fanNum);
    buffer.writeInt(packet.followNum);
    buffer.writeLong(packet.free);
    buffer.writeInt(packet.gender);
    buffer.writeLong(packet.id);
    buffer.writePacketArray(packet.items, 113);
    buffer.writePacketArray(packet.locations, 113);
    buffer.writeString(packet.name);
    buffer.writeLong(packet.normal);
    buffer.writePacketArray(packet.persons, 113);
    buffer.writeString(packet.signature);
    buffer.writeInt(packet.starNum);
};

UserCache.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new UserCache();
    const result0 = buffer.readString();
    packet.avatar = result0;
    const result1 = buffer.readString();
    packet.background = result1;
    const result2 = buffer.readLong();
    packet.coin = result2;
    const result3 = buffer.readString();
    packet.custom = result3;
    const result4 = buffer.readLong();
    packet.customTime = result4;
    const result5 = buffer.readInt();
    packet.fanNum = result5;
    const result6 = buffer.readInt();
    packet.followNum = result6;
    const result7 = buffer.readLong();
    packet.free = result7;
    const result8 = buffer.readInt();
    packet.gender = result8;
    const result9 = buffer.readLong();
    packet.id = result9;
    const list10 = buffer.readPacketArray(113);
    packet.items = list10;
    const list11 = buffer.readPacketArray(113);
    packet.locations = list11;
    const result12 = buffer.readString();
    packet.name = result12;
    const result13 = buffer.readLong();
    packet.normal = result13;
    const list14 = buffer.readPacketArray(113);
    packet.persons = list14;
    const result15 = buffer.readString();
    packet.signature = result15;
    const result16 = buffer.readInt();
    packet.starNum = result16;
    return packet;
};

export default UserCache;
